Installation issue: rocsolver-4.2.0: unable to build in internet-access restricted environment

### Steps to reproduce the issue
@arjun-raj-kuppala @srekolam

rocsolver does not install in a internet-access restricted environment even though all spack-related files were previously downloaded with `spack fetch -D rocsolver`.

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install -j100 hipblas ^cmake@3.17.5 %gcc@10.3.0                                                                                                       [+] /opt/hlrs/spack/2021-06-16/pkgconf/1.7.4-gcc-10.3.0-w6uyokeh                                                                                              [+] /opt/hlrs/spack/2021-06-16/berkeley-db/18.1.40-gcc-10.3.0-ah73t52z                                                                                        [+] /opt/hlrs/spack/2021-06-16/zlib/1.2.11-gcc-10.3.0-msipemqj                                                                                                [+] /opt/hlrs/spack/2021-06-16/libsigsegv/2.13-gcc-10.3.0-j53u2sdq                                                                                            [+] /opt/hlrs/spack/2021-06-16/util-macros/1.19.1-gcc-10.3.0-belci7vc                                                                                         [+] /opt/hlrs/spack/2021-06-16/libiconv/1.16-gcc-10.3.0-cjznarn4                                                                                              [+] /opt/hlrs/spack/2021-06-16/xz/5.2.5-gcc-10.3.0-fadnkry5                                                                                                   [+] /opt/hlrs/spack/2021-06-16/libelf/0.8.13-gcc-10.3.0-camcg7ml                                                                                              
[+] /opt/hlrs/spack/2021-06-16/libmd/1.0.3-gcc-10.3.0-6aijz37w                                                                                                
[+] /opt/hlrs/spack/2021-06-16/libffi/3.3-gcc-10.3.0-hnwxor45                                                                                                 
[+] /opt/hlrs/spack/2021-06-16/xxd-standalone/8.2.1201-gcc-10.3.0-zlfvuope                                                                                    
[+] /opt/hlrs/spack/2021-06-16/findutils/4.8.0-gcc-10.3.0-dohfaqmz                                                                                            
[+] /opt/hlrs/spack/2021-06-16/libpthread-stubs/0.4-gcc-10.3.0-r6sb24eh                                                                                       
[+] /opt/hlrs/spack/2021-06-16/xcb-proto/1.14.1-gcc-10.3.0-3sq7vbjj                                                                                           
[+] /opt/hlrs/spack/2021-06-16/libyaml/0.2.5-gcc-10.3.0-yfusswlp                                                                                              
[+] /opt/hlrs/spack/2021-06-16/util-linux-uuid/2.36.2-gcc-10.3.0-pecyjbiq                                                                                     
[+] /opt/hlrs/spack/2021-06-16/ncurses/6.2-gcc-10.3.0-yj5q2jdv                                                                                                
[+] /opt/hlrs/spack/2021-06-16/m4/1.4.19-gcc-10.3.0-hmvsex27                                                                                                  
[+] /opt/hlrs/spack/2021-06-16/inputproto/2.3.2-gcc-10.3.0-4f6pkvtk                                                                                           
[+] /opt/hlrs/spack/2021-06-16/glproto/1.4.17-gcc-10.3.0-c6fvfzyg                                                                                             
[+] /opt/hlrs/spack/2021-06-16/xtrans/1.3.5-gcc-10.3.0-4k3a34y7                                                                                               
[+] /opt/hlrs/spack/2021-06-16/xextproto/7.3.0-gcc-10.3.0-4bsxxwuf                                                                                            
[+] /opt/hlrs/spack/2021-06-16/kbproto/1.0.7-gcc-10.3.0-omtrweij                                                                                              
[+] /opt/hlrs/spack/2021-06-16/xproto/7.0.31-gcc-10.3.0-midwz2en                                                                                              
[+] /opt/hlrs/spack/2021-06-16/tar/1.34-gcc-10.3.0-mamdjjuu                                                                                                   
[+] /opt/hlrs/spack/2021-06-16/diffutils/3.7-gcc-10.3.0-stc745gd                                                                                              
[+] /opt/hlrs/spack/2021-06-16/libxml2/2.9.10-gcc-10.3.0-bm5intkz                                                                                             
[+] /opt/hlrs/spack/2021-06-16/libbsd/0.11.3-gcc-10.3.0-7vciqsoz                                                                                              
[+] /opt/hlrs/spack/2021-06-16/readline/8.1-gcc-10.3.0-nfk77ou7                                                                                               
[+] /opt/hlrs/spack/2021-06-16/libtool/2.4.6-gcc-10.3.0-4d3gj3de                                                                                              
[+] /opt/hlrs/spack/2021-06-16/libxau/1.0.8-gcc-10.3.0-3qzn34up                                                                                               
[+] /opt/hlrs/spack/2021-06-16/bzip2/1.0.8-gcc-10.3.0-hccy3l3u                                                                                                
[+] /opt/hlrs/spack/2021-06-16/expat/2.3.0-gcc-10.3.0-4u7fey6h       
[+] /opt/hlrs/spack/2021-06-16/libxdmcp/1.1.2-gcc-10.3.0-rzcjdue3
[+] /opt/hlrs/spack/2021-06-16/sqlite/3.35.5-gcc-10.3.0-5dawktfq
[+] /opt/hlrs/spack/2021-06-16/gdbm/1.19-gcc-10.3.0-4iqdf3p5
[+] /opt/hlrs/spack/2021-06-16/libpciaccess/0.16-gcc-10.3.0-xd3ibn66
[+] /opt/hlrs/spack/2021-06-16/gettext/0.21-gcc-10.3.0-ay7ju6il
[+] /opt/hlrs/spack/2021-06-16/libxcb/1.14-gcc-10.3.0-wy4ee5qn
[+] /opt/hlrs/spack/2021-06-16/perl/5.34.0-gcc-10.3.0-zztsuscd
[+] /opt/hlrs/spack/2021-06-16/hwloc/2.4.1-gcc-10.3.0-in3h2azy
[+] /opt/hlrs/spack/2021-06-16/binutils/2.36.1-gcc-10.3.0-gm6v66i7
[+] /opt/hlrs/spack/2021-06-16/bison/3.7.6-gcc-10.3.0-77ye66er
[+] /opt/hlrs/spack/2021-06-16/perl-data-dumper/2.173-gcc-10.3.0-7blne2fb
[+] /opt/hlrs/spack/2021-06-16/texinfo/6.5-gcc-10.3.0-huwafaun
[+] /opt/hlrs/spack/2021-06-16/autoconf/2.69-gcc-10.3.0-dkgirexk
[+] /opt/hlrs/spack/2021-06-16/libx11/1.7.0-gcc-10.3.0-g7dvgw4b
[+] /opt/hlrs/spack/2021-06-16/openssl/1.1.1k-gcc-10.3.0-a6eshswj
[+] /opt/hlrs/spack/2021-06-16/perl-file-which/1.22-gcc-10.3.0-pmzlqtkb
[+] /opt/hlrs/spack/2021-06-16/flex/2.6.3-gcc-10.3.0-ibuyc3xt
[+] /opt/hlrs/spack/2021-06-16/automake/1.16.3-gcc-10.3.0-fclp6mac
[+] /opt/hlrs/spack/2021-06-16/libxext/1.3.3-gcc-10.3.0-ap5mazjo
[+] /opt/hlrs/spack/2021-06-16/cmake/3.17.5-gcc-10.3.0-4kv5ctzv
[+] /opt/hlrs/spack/2021-06-16/python/3.8.10-gcc-10.3.0-b2aogdde
[+] /opt/hlrs/spack/2021-06-16/numactl/2.0.14-gcc-10.3.0-in3i3b5g
[+] /opt/hlrs/spack/2021-06-16/rocm-cmake/4.2.0-gcc-10.3.0-dhey357i
[+] /opt/hlrs/spack/2021-06-16/msgpack-c/3.1.1-gcc-10.3.0-sqbckg6o
[+] /opt/hlrs/spack/2021-06-16/py-setuptools/50.3.2-gcc-10.3.0-qbhm2y2m
[+] /opt/hlrs/spack/2021-06-16/py-pyyaml/5.3.1-gcc-10.3.0-v5dorrqr
[+] /opt/hlrs/spack/2021-06-16/hsakmt-roct/4.2.0-gcc-10.3.0-t3apqnmy
[+] /opt/hlrs/spack/2021-06-16/py-markupsafe/1.1.1-gcc-10.3.0-3u67wqw2
[+] /opt/hlrs/spack/2021-06-16/py-msgpack/1.0.2-gcc-10.3.0-ccbnf4an
[+] /opt/hlrs/spack/2021-06-16/py-wheel/0.36.2-gcc-10.3.0-6iob7ndd
[+] /opt/hlrs/spack/2021-06-16/z3/4.8.9-gcc-10.3.0-65ok556v
[+] /opt/hlrs/spack/2021-06-16/py-virtualenv/16.7.6-gcc-10.3.0-rlmhgdie
[+] /opt/hlrs/spack/2021-06-16/py-mako/1.1.4-gcc-10.3.0-s7huw5ne
[+] /opt/hlrs/spack/2021-06-16/llvm-amdgpu/4.2.0-gcc-10.3.0-vwpy3wxi
[+] /opt/hlrs/spack/2021-06-16/mesa18/18.3.6-gcc-10.3.0-yh3yhez3
[+] /opt/hlrs/spack/2021-06-16/rocm-device-libs/4.2.0-gcc-10.3.0-x3ixqt2u
[+] /opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6
[+] /opt/hlrs/spack/2021-06-16/hsa-rocr-dev/4.2.0-gcc-10.3.0-qjdhjnnn
[+] /opt/hlrs/spack/2021-06-16/rocminfo/4.2.0-gcc-10.3.0-iejqqnen
[+] /opt/hlrs/spack/2021-06-16/hip-rocclr/4.2.0-gcc-10.3.0-vtuqfmb7
[+] /opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63
[+] /opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl                                                                                     [28/1987]
==> Installing rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla
==> No binary for rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla found: installing from source
==> Warning: Suspicious requests to set or unset 'CXX' found
==> Warning:            env.set('CXX', os.path.join(link_dir, compiler.link_paths['cxx'])) at /sw/vulcan-CentOS7/hlrs/non-spack/misc/spack.2021-06-16/spack/li
b/spack/spack/build_environment.py:238
==> Warning:    --->    unknown context at unknown file:unknown line
==> Using cached archive: /localscratch/tmp/spack/hpcoft30/src/_source-cache/archive/e9/e9ef72d7c29e7c36bf02be63a64ca23b444e1ca71751749f7d66647873d9fdea.tar.g
z
==> No patches needed for rocsolver
==> rocsolver: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc-10.3.0-45lzksjw' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc-10.3.0-45lzksjw/lib;/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc-10.3.0-45lzksjw/lib64;/opt/hlrs/spack/2021-06-16/numactl/2.0.14-gcc-10.3.0-in3i3b5g/lib;/opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl/lib;/opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63/lib;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6/lib;/opt/hlrs/spack/2021-06-16/ncurses/6.2-gcc-10.3.0-yj5q2jdv/lib;/opt/hlrs/spack/2021-06-16/z3/4.8.9-gcc-10.3.0-65ok556v/lib;/opt/hlrs/spack/2021-06-16/zlib/1.2.11-gcc-10.3.0-msipemqj/lib;/opt/hlrs/spack/2021-06-16/hsa-rocr-dev/4.2.0-gcc-10.3.0-qjdhjnnn/lib;/opt/hlrs/spack/2021-06-16/libelf/0.8.13-gcc-10.3.0-camcg7ml/lib;/opt/hlrs/spack/2021-06-16/llvm-amdgpu/4.2.0-gcc-10.3.0-vwpy3wxi/lib;/opt/hlrs/spack/2021-06-16/hwloc/2.4.1-gcc-10.3.0-in3h2azy/lib;/opt/hlrs/spack/2021-06-16/libpciaccess/0.16-gcc-10.3.0-xd3ibn66/lib;/opt/hlrs/spack/2021-06-16/libxml2/2.9.10-gcc-10.3.0-bm5intkz/lib;/opt/hlrs/spack/2021-06-16/libiconv/1.16-gcc-10.3.0-cjznarn4/lib;/opt/hlrs/spack/2021-06-16/xz/5.2.5-gcc-10.3.0-fadnkry5/lib;/opt/hlrs/spack/2021-06-16/rocm-device-libs/4.2.0-gcc-10.3.0-x3ixqt2u/lib;/opt/hlrs/spack/2021-06-16/texinfo/6.5-gcc-10.3.0-huwafaun/lib;/opt/hlrs/spack/2021-06-16/perl/5.34.0-gcc-10.3.0-zztsuscd/lib;/opt/hlrs/spack/2021-06-16/berkeley-db/18.1.40-gcc-10.3.0-ah73t52z/lib;/opt/hlrs/spack/2021-06-16/gdbm/1.19-gcc-10.3.0-4iqdf3p5/lib;/opt/hlrs/spack/2021-06-16/readline/8.1-gcc-10.3.0-nfk77ou7/lib;/opt/hlrs/spack/2021-06-16/mesa18/18.3.6-gcc-10.3.0-yh3yhez3/lib;/opt/hlrs/spack/2021-06-16/expat/2.3.0-gcc-10.3.0-4u7fey6h/lib;/opt/hlrs/spack/2021-06-16/libbsd/0.11.3-gcc-10.3.0-7vciqsoz/lib;/opt/hlrs/spack/2021-06-16/libmd/1.0.3-gcc-10.3.0-6aijz37w/lib;/opt/hlrs/spack/2021-06-16/glproto/1.4.17-gcc-10.3.0-c6fvfzyg/lib;/opt/hlrs/spack/2021-06-16/libx11/1.7.0-gcc-10.3.0-g7dvgw4b/lib;/opt/hlrs/spack/2021-06-16/inputproto/2.3.2-gcc-10.3.0-4f6pkvtk/lib;/opt/hlrs/spack/2021-06-16/kbproto/1.0.7-gcc-10.3.0-omtrweij/lib;/opt/hlrs/spack/2021-06-16/libxcb/1.14-gcc-10.3.0-wy4ee5qn/lib;/opt/hlrs/spack/2021-06-16/libpthread-stubs/0.4-gcc-10.3.0-r6sb24eh/lib;/opt/hlrs/spack/2021-06-16/libxau/1.0.8-gcc-10.3.0-3qzn34up/lib;/opt/hlrs/spack/2021-06-16/xproto/7.0.31-gcc-10.3.0-midwz2en/lib;/opt/hlrs/spack/2021-06-16/libxdmcp/1.1.2-gcc-10.3.0-rzcjdue3/lib;/opt/hlrs/spack/2021-06-16/xcb-proto/1.14.1-gcc-10.3.0-3sq7vbjj/lib;/opt/hlrs/spack/2021-06-16/xextproto/7.3.0-gcc-10.3.0-4bsxxwuf/lib;/opt/hlrs/spack/2021-06-16/libxext/1.3.3-gcc-10.3.0-ap5mazjo/lib;/opt/hlrs/spack/2021-06-16/hip-rocclr/4.2.0-gcc-10.3.0-vtuqfmb7/lib;/opt/hlrs/spack/2021-06-16/msgpack-c/3.1.1-gcc-10.3.0-sqbckg6o/lib;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6/lib64;/opt/hlrs/spack/2021-06-16/hsakmt-roct/4.2.0-gcc-10.3.0-t3apqnmy/lib64' '-DCMAKE_PREFIX_PATH:STRING=/opt/hlrs/spack/2021-06-16/berkeley-db/18.1.40-gcc-10.3.0-ah73t52z;/opt/hlrs/spack/2021-06-16/cmake/3.17.5-gcc-10.3.0-4kv5ctzv;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6;/opt/hlrs/spack/2021-06-16/expat/2.3.0-gcc-10.3.0-4u7fey6h;/opt/hlrs/spack/2021-06-16/gdbm/1.19-gcc-10.3.0-4iqdf3p5;/opt/hlrs/spack/2021-06-16/glproto/1.4.17-gcc-10.3.0-c6fvfzyg;/opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63;/opt/hlrs/spack/2021-06-16/hip-rocclr/4.2.0-gcc-10.3.0-vtuqfmb7;/opt/hlrs/spack/2021-06-16/hsa-rocr-dev/4.2.0-gcc-10.3.0-qjdhjnnn;/opt/hlrs/spack/2021-06-16/hsakmt-roct/4.2.0-gcc-10.3.0-t3apqnmy;/opt/hlrs/spack/2021-06-16/hwloc/2.4.1-gcc-10.3.0-in3h2azy;/opt/hlrs/spack/2021-06-16/inputproto/2.3.2-gcc-10.3.0-4f6pkvtk;/opt/hlrs/spack/2021-06-16/kbproto/1.0.7-gcc-10.3.0-omtrweij;/opt/hlrs/spack/2021-06-16/libbsd/0.11.3-gcc-10.3.0-7vciqsoz;/opt/hlrs/spack/2021-06-16/libelf/0.8.13-gcc-10.3.0-camcg7ml;/opt/hlrs/spack/2021-06-16/libiconv/1.16-gcc-10.3.0-cjznarn4;/opt/hlrs/spack/2021-06-16/libmd/1.0.3-gcc-10.3.0-6aijz37w;/opt/hlrs/spack/2021-06-16/libpciaccess/0.16-gcc-10.3.0-xd3ibn66;/opt/hlrs/spack/2021-06-16/libpthread-stubs/0.4-gcc-10.3.0-r6sb24eh;/opt/hlrs/spack/2021-06-16/libx11/1.7.0-gcc-10.3.0-g7dvgw4b;/opt/hlrs/spack/2021-06-16/libxau/1.0.8-gcc-10.3.0-3qzn34up;/opt/hlrs/spack/2021-06-16/libxcb/1.14-gcc-10.3.0-wy4ee5qn;/opt/hlrs/spack/2021-06-16/libxdmcp/1.1.2-gcc-10.3.0-rzcjdue3;/opt/hlrs/spack/2021-06-16/libxext/1.3.3-gcc-10.3.0-ap5mazjo;/opt/hlrs/spack/2021-06-16/libxml2/2.9.10-gcc-10.3.0-bm5intkz;/opt/hlrs/spack/2021-06-16/llvm-amdgpu/4.2.0-gcc-10.3.0-vwpy3wxi;/opt/hlrs/spack/2021-06-16/mesa18/18.3.6-gcc-10.3.0-yh3yhez3;/opt/hlrs/spack/2021-06-16/msgpack-c/3.1.1-gcc-10.3.0-sqbckg6o;/opt/hlrs/spack/2021-06-16/ncurses/6.2-gcc-10.3.0-yj5q2jdv;/opt/hlrs/spack/2021-06-16/numactl/2.0.14-gcc-10.3.0-in3i3b5g;/opt/hlrs/spack/2021-06-16/perl/5.34.0-gcc-10.3.0-zztsuscd;/opt/hlrs/spack/2021-06-16/readline/8.1-gcc-10.3.0-nfk77ou7;/opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl;/opt/hlrs/spack/2021-06-16/rocm-device-libs/4.2.0-gcc-10.3.0-x3ixqt2u;/opt/hlrs/spack/2021-06-16/texinfo/6.5-gcc-10.3.0-huwafaun;/opt/hlrs/spack/2021-06-16/xcb-proto/1.14.1-gcc-10.3.0-3sq7vbjj;/opt/hlrs/spack/2021-06-16/xextproto/7.3.0-gcc-10.3.0-4bsxxwuf;/opt/hlrs/spack/2021-06-16/xproto/7.0.31-gcc-10.3.0-midwz2en;/opt/hlrs/spack/2021-06-16/xtrans/1.3.5-gcc-10.3.0-4k3a34y7;/opt/hlrs/spack/2021-06-16/xz/5.2.5-gcc-10.3.0-fadnkry5;/opt/hlrs/spack/2021-06-16/z3/4.8.9-gcc-10.3.0-65ok556v;/opt/hlrs/spack/2021-06-16/zlib/1.2.11-gcc-10.3.0-msipemqj' '-DBUILD_CLIENTS_SAMPLES=OFF' '-DBUILD_CLIENTS_TESTS=OFF' '-DBUILD_CLIENTS_BENCHMARKS=OFF' '-DCMAKE_CXX_FLAGS:String=-I/opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl/rocblas/include' '-DOPTIMAL:BOOL=ON' '/localscratch/tmp/spack/hpcoft30/build/spack-stage-rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla/spack-src'

1 error found in build log:
     12    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
     13    -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
     14    -- Check if compiler accepts -pthread
     15    -- Check if compiler accepts -pthread - yes
     16    -- Found Threads: TRUE
     17    -- Use hip-clang to build for amdgpu backend
  >> 18    CMake Error at cmake/get-rocm-cmake.cmake:25 (message):
     19      error: downloading
     20    
     21      'https://github.com/RadeonOpenCompute/rocm-cmake/archive/master.zip' failed
     22    
     23          status_code: 7
     24          status_string: "Couldn't connect to server"

See build log for details:
  /localscratch/tmp/spack/hpcoft30/build/spack-stage-rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla/spack-build-out.txt

==> Warning: Skipping build of hipblas-4.2.0-kcnufctmgpvftcftnlwni7t3qd3xbnr7 since rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla failed
==> Error: hipblas-4.2.0-kcnufctmgpvftcftnlwni7t3qd3xbnr7: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

### Information on your system

<!-- Please include the output of `spack debug report` -->
```console
$ spack debug report
* **Spack:** 0.16.2-3143-64f3e37
* **Python:** 3.6.8
* **Platform:** linux-centos7-zen2
* **Concretizer:** original
```


<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists. -->
See below for `spack-build-out.txt` and `spack-build-env.txt`.

<!-- Some packages have maintainers who have volunteered to debug build failures. Run `spack maintainers <name-of-the-package>` and @mention them here if they exist. -->

### General information

<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

`spack-build-out.txt`
```
==> rocsolver: Executing phase: 'cmake' 
==> [2021-06-17-13:45:51.618834] 'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc-10.3.0-45lzksjw' 
'-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_
PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc-10.3.0-45lzksjw/lib;/opt/hlrs/spack/2021-06-16/rocsolver/4.2.0-gcc
-10.3.0-45lzksjw/lib64;/opt/hlrs/spack/2021-06-16/numactl/2.0.14-gcc-10.3.0-in3i3b5g/lib;/opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl/lib;/opt
/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63/lib;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6/lib;/opt/hlrs/spack/2021-06-16/ncurses/6.2
-gcc-10.3.0-yj5q2jdv/lib;/opt/hlrs/spack/2021-06-16/z3/4.8.9-gcc-10.3.0-65ok556v/lib;/opt/hlrs/spack/2021-06-16/zlib/1.2.11-gcc-10.3.0-msipemqj/lib;/opt/hlrs/
spack/2021-06-16/hsa-rocr-dev/4.2.0-gcc-10.3.0-qjdhjnnn/lib;/opt/hlrs/spack/2021-06-16/libelf/0.8.13-gcc-10.3.0-camcg7ml/lib;/opt/hlrs/spack/2021-06-16/llvm-a
mdgpu/4.2.0-gcc-10.3.0-vwpy3wxi/lib;/opt/hlrs/spack/2021-06-16/hwloc/2.4.1-gcc-10.3.0-in3h2azy/lib;/opt/hlrs/spack/2021-06-16/libpciaccess/0.16-gcc-10.3.0-xd3
ibn66/lib;/opt/hlrs/spack/2021-06-16/libxml2/2.9.10-gcc-10.3.0-bm5intkz/lib;/opt/hlrs/spack/2021-06-16/libiconv/1.16-gcc-10.3.0-cjznarn4/lib;/opt/hlrs/spack/2
021-06-16/xz/5.2.5-gcc-10.3.0-fadnkry5/lib;/opt/hlrs/spack/2021-06-16/rocm-device-libs/4.2.0-gcc-10.3.0-x3ixqt2u/lib;/opt/hlrs/spack/2021-06-16/texinfo/6.5-gc
c-10.3.0-huwafaun/lib;/opt/hlrs/spack/2021-06-16/perl/5.34.0-gcc-10.3.0-zztsuscd/lib;/opt/hlrs/spack/2021-06-16/berkeley-db/18.1.40-gcc-10.3.0-ah73t52z/lib;/o
pt/hlrs/spack/2021-06-16/gdbm/1.19-gcc-10.3.0-4iqdf3p5/lib;/opt/hlrs/spack/2021-06-16/readline/8.1-gcc-10.3.0-nfk77ou7/lib;/opt/hlrs/spack/2021-06-16/mesa18/1
8.3.6-gcc-10.3.0-yh3yhez3/lib;/opt/hlrs/spack/2021-06-16/expat/2.3.0-gcc-10.3.0-4u7fey6h/lib;/opt/hlrs/spack/2021-06-16/libbsd/0.11.3-gcc-10.3.0-7vciqsoz/lib;
/opt/hlrs/spack/2021-06-16/libmd/1.0.3-gcc-10.3.0-6aijz37w/lib;/opt/hlrs/spack/2021-06-16/glproto/1.4.17-gcc-10.3.0-c6fvfzyg/lib;/opt/hlrs/spack/2021-06-16/li
bx11/1.7.0-gcc-10.3.0-g7dvgw4b/lib;/opt/hlrs/spack/2021-06-16/inputproto/2.3.2-gcc-10.3.0-4f6pkvtk/lib;/opt/hlrs/spack/2021-06-16/kbproto/1.0.7-gcc-10.3.0-omt
rweij/lib;/opt/hlrs/spack/2021-06-16/libxcb/1.14-gcc-10.3.0-wy4ee5qn/lib;/opt/hlrs/spack/2021-06-16/libpthread-stubs/0.4-gcc-10.3.0-r6sb24eh/lib;/opt/hlrs/spa
ck/2021-06-16/libxau/1.0.8-gcc-10.3.0-3qzn34up/lib;/opt/hlrs/spack/2021-06-16/xproto/7.0.31-gcc-10.3.0-midwz2en/lib;/opt/hlrs/spack/2021-06-16/libxdmcp/1.1.2-
gcc-10.3.0-rzcjdue3/lib;/opt/hlrs/spack/2021-06-16/xcb-proto/1.14.1-gcc-10.3.0-3sq7vbjj/lib;/opt/hlrs/spack/2021-06-16/xextproto/7.3.0-gcc-10.3.0-4bsxxwuf/lib
;/opt/hlrs/spack/2021-06-16/libxext/1.3.3-gcc-10.3.0-ap5mazjo/lib;/opt/hlrs/spack/2021-06-16/hip-rocclr/4.2.0-gcc-10.3.0-vtuqfmb7/lib;/opt/hlrs/spack/2021-06-
16/msgpack-c/3.1.1-gcc-10.3.0-sqbckg6o/lib;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6/lib64;/opt/hlrs/spack/2021-06-16/hsakmt-roct/4.2.0-gcc-1
0.3.0-t3apqnmy/lib64' '-DCMAKE_PREFIX_PATH:STRING=/opt/hlrs/spack/2021-06-16/berkeley-db/18.1.40-gcc-10.3.0-ah73t52z;/opt/hlrs/spack/2021-06-16/cmake/3.17.5-g
cc-10.3.0-4kv5ctzv;/opt/hlrs/spack/2021-06-16/comgr/4.2.0-gcc-10.3.0-l325zvp6;/opt/hlrs/spack/2021-06-16/expat/2.3.0-gcc-10.3.0-4u7fey6h;/opt/hlrs/spack/2021-
06-16/gdbm/1.19-gcc-10.3.0-4iqdf3p5;/opt/hlrs/spack/2021-06-16/glproto/1.4.17-gcc-10.3.0-c6fvfzyg;/opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63;/op
t/hlrs/spack/2021-06-16/hip-rocclr/4.2.0-gcc-10.3.0-vtuqfmb7;/opt/hlrs/spack/2021-06-16/hsa-rocr-dev/4.2.0-gcc-10.3.0-qjdhjnnn;/opt/hlrs/spack/2021-06-16/hsak
mt-roct/4.2.0-gcc-10.3.0-t3apqnmy;/opt/hlrs/spack/2021-06-16/hwloc/2.4.1-gcc-10.3.0-in3h2azy;/opt/hlrs/spack/2021-06-16/inputproto/2.3.2-gcc-10.3.0-4f6pkvtk;/
opt/hlrs/spack/2021-06-16/kbproto/1.0.7-gcc-10.3.0-omtrweij;/opt/hlrs/spack/2021-06-16/libbsd/0.11.3-gcc-10.3.0-7vciqsoz;/opt/hlrs/spack/2021-06-16/libelf/0.8
.13-gcc-10.3.0-camcg7ml;/opt/hlrs/spack/2021-06-16/libiconv/1.16-gcc-10.3.0-cjznarn4;/opt/hlrs/spack/2021-06-16/libmd/1.0.3-gcc-10.3.0-6aijz37w;/opt/hlrs/spac
k/2021-06-16/libpciaccess/0.16-gcc-10.3.0-xd3ibn66;/opt/hlrs/spack/2021-06-16/libpthread-stubs/0.4-gcc-10.3.0-r6sb24eh;/opt/hlrs/spack/2021-06-16/libx11/1.7.0
-gcc-10.3.0-g7dvgw4b;/opt/hlrs/spack/2021-06-16/libxau/1.0.8-gcc-10.3.0-3qzn34up;/opt/hlrs/spack/2021-06-16/libxcb/1.14-gcc-10.3.0-wy4ee5qn;/opt/hlrs/spack/20
21-06-16/libxdmcp/1.1.2-gcc-10.3.0-rzcjdue3;/opt/hlrs/spack/2021-06-16/libxext/1.3.3-gcc-10.3.0-ap5mazjo;/opt/hlrs/spack/2021-06-16/libxml2/2.9.10-gcc-10.3.0-
bm5intkz;/opt/hlrs/spack/2021-06-16/llvm-amdgpu/4.2.0-gcc-10.3.0-vwpy3wxi;/opt/hlrs/spack/2021-06-16/mesa18/18.3.6-gcc-10.3.0-yh3yhez3;/opt/hlrs/spack/2021-06
-16/msgpack-c/3.1.1-gcc-10.3.0-sqbckg6o;/opt/hlrs/spack/2021-06-16/ncurses/6.2-gcc-10.3.0-yj5q2jdv;/opt/hlrs/spack/2021-06-16/numactl/2.0.14-gcc-10.3.0-in3i3b
5g;/opt/hlrs/spack/2021-06-16/perl/5.34.0-gcc-10.3.0-zztsuscd;/opt/hlrs/spack/2021-06-16/readline/8.1-gcc-10.3.0-nfk77ou7;/opt/hlrs/spack/2021-06-16/rocblas/4
.2.0-gcc-10.3.0-widki3gl;/opt/hlrs/spack/2021-06-16/rocm-device-libs/4.2.0-gcc-10.3.0-x3ixqt2u;/opt/hlrs/spack/2021-06-16/texinfo/6.5-gcc-10.3.0-huwafaun;/opt
/hlrs/spack/2021-06-16/xcb-proto/1.14.1-gcc-10.3.0-3sq7vbjj;/opt/hlrs/spack/2021-06-16/xextproto/7.3.0-gcc-10.3.0-4bsxxwuf;/opt/hlrs/spack/2021-06-16/xproto/7
.0.31-gcc-10.3.0-midwz2en;/opt/hlrs/spack/2021-06-16/xtrans/1.3.5-gcc-10.3.0-4k3a34y7;/opt/hlrs/spack/2021-06-16/xz/5.2.5-gcc-10.3.0-fadnkry5;/opt/hlrs/spack/
2021-06-16/z3/4.8.9-gcc-10.3.0-65ok556v;/opt/hlrs/spack/2021-06-16/zlib/1.2.11-gcc-10.3.0-msipemqj' '-DBUILD_CLIENTS_SAMPLES=OFF' '-DBUILD_CLIENTS_TESTS=OFF' 
'-DBUILD_CLIENTS_BENCHMARKS=OFF' '-DCMAKE_CXX_FLAGS:String=-I/opt/hlrs/spack/2021-06-16/rocblas/4.2.0-gcc-10.3.0-widki3gl/rocblas/include' '-DOPTIMAL:BOOL=ON'
 '/localscratch/tmp/spack/hpcoft30/build/spack-stage-rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla/spack-src'
-- The CXX compiler identification is Clang 12.0.0
-- Check for working CXX compiler: /opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63/bin/hipcc
-- Check for working CXX compiler: /opt/hlrs/spack/2021-06-16/hip/4.2.0-gcc-10.3.0-lnphjz63/bin/hipcc - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Use hip-clang to build for amdgpu backend
CMake Error at cmake/get-rocm-cmake.cmake:25 (message):
  error: downloading

  'https://github.com/RadeonOpenCompute/rocm-cmake/archive/master.zip' failed

      status_code: 7
      status_string: "Couldn't connect to server"
      log:   Trying 140.82.121.3:443... 

  TCP_NODELAY set

  connect to 140.82.121.3 port 443 failed: Connection timed out

  Failed to connect to github.com port 443: Connection timed out

  Closing connection 0

   

Call Stack (most recent call first):
  CMakeLists.txt:101 (include)


-- Configuring incomplete, errors occurred!
See also "/localscratch/tmp/spack/hpcoft30/build/spack-stage-rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla/spack-build-45lzksj/CMakeFiles/CMakeOutput.log".
See also "/localscratch/tmp/spack/hpcoft30/build/spack-stage-rocsolver-4.2.0-45lzksjwehfivrbtuqzaskj74unxllla/spack-build-45lzksj/CMakeFiles/CMakeError.log".

--- a/var/spack/repos/builtin/packages/hipcub/package.py
+++ b/var/spack/repos/builtin/packages/hipcub/package.py
@@ -33,6 +33,7 @@ class Hipcub(CMakePackage):
                 '4.2.0']:
         depends_on('hip@' + ver, when='@' + ver)
         depends_on('rocprim@' + ver, when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     for ver in ['4.1.0', '4.2.0']:
         depends_on('hip-rocclr@' + ver, when='@' + ver)

--- a/var/spack/repos/builtin/packages/hipfort/package.py
+++ b/var/spack/repos/builtin/packages/hipfort/package.py
@@ -28,6 +28,7 @@ class Hipfort(CMakePackage):
 
     for ver in ['3.8.0', '3.9.0', '3.10.0', '4.0.0', '4.1.0', '4.2.0']:
         depends_on('hip@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     def setup_build_environment(self, env):
         env.set('CXX', self.spec['hip'].hipcc)

--- a/var/spack/repos/builtin/packages/rocalution/package.py
+++ b/var/spack/repos/builtin/packages/rocalution/package.py
@@ -39,6 +39,7 @@ class Rocalution(CMakePackage):
         depends_on('rocsparse@' + ver, when='@' + ver)
         depends_on('comgr@' + ver, when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     for ver in ['3.9.0', '3.10.0', '4.0.0', '4.1.0', '4.2.0']:
         depends_on('rocrand@' + ver, when='@' + ver)

--- a/var/spack/repos/builtin/packages/rocprim/package.py
+++ b/var/spack/repos/builtin/packages/rocprim/package.py
@@ -35,6 +35,7 @@ class Rocprim(CMakePackage):
         depends_on('comgr@' + ver, when='@' + ver)
         depends_on('hsa-rocr-dev@' + ver, when='@' + ver)
         depends_on('llvm-amdgpu@' + ver, when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     for ver in ['4.1.0', '4.2.0']:
         depends_on('hip-rocclr@' + ver, type='build', when='@' + ver)

--- a/var/spack/repos/builtin/packages/rocrand/package.py
+++ b/var/spack/repos/builtin/packages/rocrand/package.py
@@ -38,6 +38,7 @@ class Rocrand(CMakePackage):
         depends_on('llvm-amdgpu@' + ver, type='build', when='@' + ver)
         depends_on('rocminfo@' + ver, type='build', when='@' + ver)
         depends_on('hsa-rocr-dev@' + ver, type='build', when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
     for ver in ['4.1.0', '4.2.0']:
         depends_on('hip-rocclr@' + ver, type='link', when='@' + ver)
 

--- a/var/spack/repos/builtin/packages/rocsolver/package.py
+++ b/var/spack/repos/builtin/packages/rocsolver/package.py
@@ -41,6 +41,7 @@ class Rocsolver(CMakePackage):
         depends_on('hip@' + ver, type='build', when='@' + ver)
         depends_on('comgr@' + ver, type='build', when='@' + ver)
         depends_on('rocblas@' + ver, type='link', when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     def cmake_args(self):
         args = [

--- a/var/spack/repos/builtin/packages/rocsparse/package.py
+++ b/var/spack/repos/builtin/packages/rocsparse/package.py
@@ -34,6 +34,7 @@ class Rocsparse(CMakePackage):
                 '4.2.0']:
         depends_on('hip@' + ver, when='@' + ver)
         depends_on('rocprim@' + ver, when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     for ver in ['4.1.0', '4.2.0']:
         depends_on('hip-rocclr@' + ver, when='@' + ver)

--- a/var/spack/repos/builtin/packages/rocthrust/package.py
+++ b/var/spack/repos/builtin/packages/rocthrust/package.py
@@ -37,6 +37,7 @@ class Rocthrust(CMakePackage):
                 '4.2.0']:
         depends_on('hip@' + ver, when='@' + ver)
         depends_on('rocprim@' + ver, when='@' + ver)
+        depends_on('rocm-cmake@' + ver, type='build', when='@' + ver)
 
     for ver in ['4.1.0', '4.2.0']:
         depends_on('hip-rocclr@' + ver, when='@' + ver)

